src/app.ts

```diff
@@ -166,7 +164,34 @@ export class App extends Construct {
       default:
         break;
     }
+
+  }
+
+  /**
+   * Synthesizes the app into a YAML string.
+   *
+   * @returns A string with all YAML objects across all charts in this app.
+   */
+  public synthYaml(): any {
+    // Since we plan on removing the distributed synth mechanism, we no longer call `Node.synthesize`, but rather simply implement
+    // the necessary operations. We do however want to preserve the distributed validation.
+    validate(this);
+
+    var str = ''; // string we will concatenate all the yaml objects into
```

Comment:
Collect each `Yaml.stringify()` result into an array of strings and then `.join('---\n')` them together before returning.